src/main/java/com/example/animalshelter/controller/AnimalController.java

```diff
@@ -0,0 +1,22 @@
+package com.example.animalshelter.controller;
+
+import com.example.animalshelter.model.Animal;
+import com.example.animalshelter.service.AnimalService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController()
+public class AnimalController {
+
+    private final AnimalService animalService;
+
+    public AnimalController(AnimalService animalService) {
+        this.animalService = animalService;
+    }
+
+    @PostMapping(value = "/save", consumes = "application/json", produces = "application/json")
+    public Animal saveAnimal(@RequestBody Animal animal) {
+        return animalService.save(animal) ? animal : new Animal();
```

Comment:
I think we should return the status code. Ok -201, fail - 400 ?

Reply:
Now we return status 200 if request is resolved successfully and 400 if something go wrong, also there is string response.